File: backend/src/main/java/com/service/backend/mentorship/entity/MenteeProfile.java
package com.service.backend.mentorship.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.annotation.Transient;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("mentee_profiles")
public class MenteeProfile implements Persistable<Integer> {

    @Id
    @Column("member_id")
    private Integer memberId;

    @Column("mentoring_goal")
    private String mentoringGoal;

    @Column("major")
    private String major;

    @Column("academic_year")
    private String academicYear;

    @Column("interests")
    private String interests;

    @Column("is_active")
    @Builder.Default
    private Boolean isActive = true;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column("updated_at")
    private LocalDateTime updatedAt;

    @Transient
    @Builder.Default
    private boolean isNew = true;

    @Override
    public Integer getId() {
        return memberId;
    }

    @Override
    public boolean isNew() {
        return isNew;
    }
}

File: mobile_flutter/lib/features/mentorship/data/models/mentee_profile.dart

class MenteeProfile {
  final int memberId;
  final String mentoringGoal;
  final String major;
  final String academicYear;
  final String? interests;
  final bool isActive;

  const MenteeProfile({
    required this.memberId,
    required this.mentoringGoal,
    required this.major,
    required this.academicYear,
    this.interests,
    this.isActive = true,
  });

  factory MenteeProfile.fromJson(Map<String, dynamic> json) {
    return MenteeProfile(
      memberId: (json['memberId'] as num).toInt(),
      mentoringGoal: json['mentoringGoal'] as String? ?? '',
      major: json['major'] as String? ?? '',
      academicYear: json['academicYear'] as String? ?? '',
      interests: json['interests'] as String?,
      isActive: json['isActive'] as bool? ?? true,
    );
  }
}
